compiler/rustc_hir_analysis/src/hir_ty_lowering/errors.rs

```diff
@@ -1047,23 +1047,27 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
             };
         });
 
+        let segments: Vec<_> = match err_extend {
+            GenericsArgsErrExtend::DefVariant(segments) => segments.iter().collect(),
+            _ => segments.collect(),
+        };
```

Comment:
I think I agree with lcnr about how having extra spans and things in the error message for the enum arguments too is confusing. Can you just remove this so we keep the current set of segments, then I think this is good to merge

Reply:
done